onnxruntime/core/providers/cuda/cuda_stream_handle.cc

```diff
@@ -4,6 +4,36 @@
 
 namespace onnxruntime {
 
+    struct StreamPool {
+        ~StreamPool() {
+            for (const auto& s : streams_) {
+                CUDA_CALL_THROW(cudaStreamDestroy(s));
+            }
+        }
+        cudaStream_t GetStream() {
+          if (streams_.empty()) {
+            cudaStream_t stream;
+            CUDA_CALL_THROW(cudaStreamCreateWithFlags(&stream, cudaStreamNonBlocking));
+            return stream;
+          } else {
+            cudaStream_t stream = streams_.back();
+            streams_.pop_back();
+			return stream;
+          }
+        }
+		
+        void PutStream(cudaStream_t stream) {
+          ORT_ENFORCE(stream);
```

Comment:
Since the PuStream will be called in destructor, we might want to stop using ORT_ENFORCE here since the exception in destructor will cause instant process termination.

Reply:
+1, if you want to avoid nullptr, just do a check with if.